src/metabase/upload.clj

```diff
@@ -199,26 +199,30 @@
   [s]
   (str/replace s currency-regex ""))
 
-(defn- parse-plain-number [s]
-  (case (get-number-separators)
-    ("." ".,") (. (NumberFormat/getInstance (Locale. "en" "US")) parse s)
-    ",." (. (NumberFormat/getInstance (Locale. "de" "DE")) parse s)
-    ", " (. (NumberFormat/getInstance (Locale. "fr" "FR")) parse (str/replace s \space \u00A0)) ; \u00A0 is a non-breaking space
-    ".’" (. (NumberFormat/getInstance (Locale. "de" "CH")) parse s)))
+(let [us (NumberFormat/getInstance (Locale. "en" "US"))
+      de (NumberFormat/getInstance (Locale. "de" "DE"))
+      fr (NumberFormat/getInstance (Locale. "fr" "FR"))
+      ch (NumberFormat/getInstance (Locale. "de" "CH"))]
```

Comment:
memory cost of keeping these around forever is trivial, and it was a major speedup not to instantiate a new `Locale` and call `getInstance` for every single number that needed to be parsed.